### DISCUSSION TITLE
Confused with alembic + SQLAlchemy Base.metadata

### DISCUSSION THREAD

**Discussion by kernelguardian** [sqlalchemy/alembic#960] [Usage Questions] [answered]
For using alembic autogenerate feature it is mentioned in the documentation to import the `Base` from models and add the `Base.metadata` attribute to a target variable.  But when I tried importing the model class directly say `myModelClass` and assigning that to the target variable instead of `Base.metadata` , i found no difference. Later I tried importing models in other files and they were also detected by alembic without adding them to a list. what exactly does alembic do with the Base.metadata attribute.

**Comment by CaselIT** [comment, marked as answer]
I suggest reading the fist sections of this documentation page https://alembic.sqlalchemy.org/en/latest/autogenerate.html#auto-generating-migrations

TLDR: it compares what's in the metadata with was is in the db and writes the difference in the generated revision

**Reply by kernelguardian** [reply]
I did read those, I had some idea about Base.metadata, forgive me for asking but how is Base.metadata different from say myModel.metadata.

**Reply by CaselIT** [reply]
I think `Base` is just referring to the example. what alembic looks for is one (or more) metadata instances, and usually when using the orm with declarative meta the base class of all models is called `Base` in the examples. But any metadata is fine there

**Reply by kernelguardian** [reply]
So say I have two models `modelA` in fileA.py and `modelB` in fileB.py. Do I have to assign `target_metadata = [modelA, modelB]` like this or `target_metadata = Base` or `target_metadata = modelA`.

**Reply by CaselIT** [reply]
Please note that with metadata we are talking about https://docs.sqlalchemy.org/en/14/core/metadata.html instances, not models.

And Base in the example is defined like this https://docs.sqlalchemy.org/en/14/orm/mapping_styles.html#declarative-mapping

**Reply by kernelguardian** [reply]
Sorry I made a mistake there , 
So say I have two models modelA in fileA.py and modelB in fileB.py. Do I have to assign `target_metadata = [modelA.Base.metadata, modelB.Base.metadata]` like this or `target_metadata = Base.metadata` or `target_metadata = modelA.Base.metadata`. does this makes sense?

> So say I have two models `modelA` in fileA.py and `modelB` in fileB.py. Do I have to assign `target_metadata = [modelA, modelB]` like this or `target_metadata = Base` or `target_metadata = modelA`.

**Reply by CaselIT** [reply]
Assuming that the models are defined as
```py
class modelA(Base):
   ...
...
class modelB(Base):
   ...
```
you would use `Base.metadata`
